### ISSUE TITLE
svtplay-dl on Synology NAS systems?

### ISSUE THREAD

**Issue by hereinmyhead** [spaam/svtplay-dl#1415]
Hi,

It isn't really an issue but I wonder if anyone know if it is possible to run svtplay-dl on a Synology NAS? If it is possible, any way to point me to the right direction?  I have basic computer knowledge but no programing skills more than basic webpages. Maybe it is to hard to do for a non programmer? 

Thank you!

**Comment by spaam** [comment]
I have no idea how the synology nas thing work. but if you got shell access to it you can check if you have python 3.6+ installed? if so you could use the svtplay-dl binary thing but you have to install some dependencies to use it :)

**Comment by Nionor** [comment]
First it depends on witch one you have, if it's fairly new it should work fine.
You need to have SSH access and then you can just follow the instructions "For any Linux". (I can be done with the task scheduler too but it's a bit more complicated.)

**Comment by hereinmyhead** [comment]
I have a Synology 1815+. I think I've succeded with installation using SSH. I installed svtplay-dl using pip3. My problem now is that I can't upgrade to svtplay-dl 4.4 that was released today. I use the `sudo pip3 install svtplay-dl --upgrade` command and get this:

> sudo pip3 install svtplay-dl --upgrade
Requirement already up-to-date: svtplay-dl in /usr/lib/python3.8/site-packages (4.4)
Requirement already satisfied, skipping upgrade: cryptography in /usr/lib/python3.8/site-packages (from svtplay-dl) (3.4.8)
Requirement already satisfied, skipping upgrade: PySocks in /usr/lib/python3.8/site-packages (from svtplay-dl) (1.7.1)
Requirement already satisfied, skipping upgrade: pyyaml in /usr/lib/python3.8/site-packages (from svtplay-dl) (5.4.1)
Requirement already satisfied, skipping upgrade: requests>=2.0.0 in /usr/lib/python3.8/site-packages (from svtplay-dl) (2.26.0)
Requirement already satisfied, skipping upgrade: cffi>=1.12 in /usr/lib/python3.8/site-packages (from cryptography->svtplay-dl) (1.14.6)
Requirement already satisfied, skipping upgrade: charset-normalizer~=2.0.0; python_version >= "3" in /usr/lib/python3.8/site-packages (from requests>=2.0.0->svtplay-dl) (2.0.4)
Requirement already satisfied, skipping upgrade: certifi>=2017.4.17 in /usr/lib/python3.8/site-packages (from requests>=2.0.0->svtplay-dl) (2021.5.30)
Requirement already satisfied, skipping upgrade: idna<4,>=2.5; python_version >= "3" in /usr/lib/python3.8/site-packages (from requests>=2.0.0->svtplay-dl) (3.2)
Requirement already satisfied, skipping upgrade: urllib3<1.27,>=1.21.1 in /usr/lib/python3.8/site-packages (from requests>=2.0.0->svtplay-dl) (1.26.6)
Requirement already satisfied, skipping upgrade: pycparser in /usr/lib/python3.8/site-packages (from cffi>=1.12->cryptography->svtplay-dl) (2.20)

When I run `svtplay-dl --version` i get svtplay-dl 4.3 and not 4.4.

Thank you for your help!

**Comment by Nionor** [comment]
I've not used the python one but maybe it's that you have different versions installed for your regular user and your sudo user? Not sure if that could happen with python.
Try `sudo svtplay-dl --version` and see if that makes a difference.

**Comment by hereinmyhead** [comment]
After some digging I found that I had two versions of svtplay-dl - one with pip3 and one at another location (don't ask me how I did it :)). I deleted both installations manually and installed with pip3 again. Now it's working!

Thank you!

**Comment by annieran** [comment]
On my Synology I’m running svtplay-dl in a ubuntu container using the Synology docker app(not supported on all Synology nas). 
I have a folder map to container where all the video ends up. 
One  big advantages with this setup is that a can access it with docker app in Synology DSM web interface locally and remotely (quickconnect or vpn)  without open any ports in my router.

**Comment by hereinmyhead** [comment]
@annieran: Thank you! Are you using the official ubuntu docker container? (https://registry.hub.docker.com/_/ubuntu/). I've tried with that one but it seems that you have to manually install some packages that I dont't understand at this moment...

**Comment by annieran** [comment]
@hereinmyhead Yes, I using the official ubuntu, what I can remember from when I installing it a was falling the install guide for "install svt-dl on ubuntu” on the svtplay-dl website (or github page). Without any problem.  Have you install the container and follow the install guide?
